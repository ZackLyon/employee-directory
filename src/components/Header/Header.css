header {
  height: 10vh;
  width: 100vw;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: solid black 1px;
}

header > * {
  margin-left: 3vmax;
  margin-right: 3vmax;
}

.signoutContainer {
  display: flex;
  align-items: center;
}

.signoutContainer > button {
  height: fit-content;
  margin-left: 1vmax;
}
