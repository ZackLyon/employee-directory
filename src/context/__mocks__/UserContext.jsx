import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ children, testUser }) => {
  const [user, setUser] = useState({ ...testUser });
  //user from server is this obj: {id: x, email: xxx}

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error(
      'useUser can only be used on a descendant element of the Provider'
    );
  }

  return context;
};

export { UserProvider, useUser };
