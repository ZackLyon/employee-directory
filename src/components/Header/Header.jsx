import { Link, useHistory } from 'react-router-dom';
import { useUser } from '../../context/UserContext.jsx';
import styles from './Header.css';

export default function Header() {
  const { user, setUser } = useUser();
  const history = useHistory();

  const handleClick = () => {
    setUser({});
    history.replace('/login');
  };

  return (
    <header>
      <Link to="/">
        <h1>Acme Inc.</h1>
      </Link>
      <div>
        {user.id ? (
          <div className={styles.signoutContainer}>
            <h3>{user.email}</h3>
            <button onClick={handleClick}>Sign Out</button>
          </div>
        ) : (
          <Link to="/login">
            <button>Log In</button>
          </Link>
        )}
      </div>
    </header>
  );
}
