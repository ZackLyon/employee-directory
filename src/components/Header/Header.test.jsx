import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext.jsx';
import Header from './Header.jsx';

jest.mock('../../context/UserContext.jsx');

it('should display a Log In button if no user is logged in', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider testUser={{}}>
        <Header />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

it("should display a user's email if logged in", () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider testUser={{ id: '1234', email: 'some@guy.com' }}>
        <Header />
      </UserProvider>
    </MemoryRouter>
  );

  screen.getByText('some@guy.com');
  expect(container).toMatchSnapshot();
});
