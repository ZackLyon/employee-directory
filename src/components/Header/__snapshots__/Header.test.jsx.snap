// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a Log In button if no user is logged in 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      <h1>
        Acme Inc.
      </h1>
    </a>
    <div>
      <a
        href="/login"
      >
        <button>
          Log In
        </button>
      </a>
    </div>
  </header>
</div>
`;

exports[`should display a user's email if logged in 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      <h1>
        Acme Inc.
      </h1>
    </a>
    <div>
      <div
        class="signoutContainer"
      >
        <h3>
          some@guy.com
        </h3>
        <button>
          Sign Out
        </button>
      </div>
    </div>
  </header>
</div>
`;
